### ISSUE TITLE
Experiment: k-mer tokenization for promoters

### ISSUE THREAD

**Issue by gonzalobenegas** [Open-Athena/bolinas-dna#64]
## Description

Instead of character-level tokenization, try https://huggingface.co/bolinas-dna/tokenizer-4-mer and https://huggingface.co/bolinas-dna/tokenizer-8-mer. Train on promoter dataset and evaluate on zero-shot VEP.

## Hypothesis or Goal

Tokenization can influence both downstream task performance and training/inference speed. Character-level is a good default but worth exploring additional options.

## Links

[Training code](https://github.com/marin-community/marin/blob/f55f33252112700d78c432307d2e7ceee214265f/experiments/dna/exp64_kmer_tokenization.py)
[Analysis code](https://github.com/Open-Athena/bolinas-dna/tree/main/snakemake/analysis/evals_v1)
Wandb: [baseline](https://wandb.ai/gonzalobenegas/marin/runs/exp55-mammals-r01-52f413?nw=nwusergonzalobenegas), [4-mer](https://wandb.ai/gonzalobenegas/marin/runs/exp64-genomes-v4-genome_set-mammals-intervals-v1_256_128-4me-r01-0ed540?nw=nwusergonzalobenegas), [8-mer](https://wandb.ai/gonzalobenegas/marin/runs/exp64-genomes-v4-genome_set-mammals-intervals-v1_256_128-8me-r01-eee19b?nw=nwusergonzalobenegas)

## Results

Eval: TraitGym Mendelian traits v2 (validation set, only odd chromosomes).
- With increasing $k$, $k$-mer tokenization degrades performance.
- Small caveat: the baseline run was trained with repeat downweighting #9 but not the k-mer models. Still, I don't think this would explain such differences (and does not affect the comparison between 4-mer and 8-mer).
<img width="1305" height="509" alt="Image" src="https://github.com/user-attachments/assets/136c696f-b779-4185-852e-fb6e85aaa80a" />
